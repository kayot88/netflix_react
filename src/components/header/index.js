import React, { useState } from 'react';
// import SearchIcon from '@material-ui/icons/Search';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import { Link as ReachRouterLink } from 'react-router-dom';
import {
  Container,
  Background,
  Picture,
  Link,
  FeatureCallOut,
  ButtonLink,
  Text,
  Feature,
  Logo,
  PlayButton,
  Group,
  Profile,
  Dropdown,
  Search,
  SearchIcon,
  SearchInput,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.PlayButton = (({ children, ...restProps}) => {
return <PlayButton {...restProps}>{children}</PlayButton>
}  )

Header.Search = function HeaderSearch({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);
  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png"  alt="Search"/>
      </SearchIcon>
      <SearchInput
        active={searchActive}
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
      />
    </Search>
  );
};
Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};
Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReachRouterLink to={to}>
      <Logo {...restProps} />
    </ReachRouterLink>
  );
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature>{children}</Feature>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/users/1.png`}
    />
  );
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.Link = function HeaderLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};
