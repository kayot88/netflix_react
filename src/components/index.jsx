export { default as Jumbotron } from "./jumbo/jumbotron";
export { default as Footer } from "./footer/footer";
export { default as Accordeon } from "./accordeon/accordeon";
export { default as OptForm } from "./optForm/optForm";
export { default as HeaderComponent } from "./header/header";
export { default as Form } from "./form/form";
