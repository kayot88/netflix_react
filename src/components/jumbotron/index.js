import React from 'react';
import {
  Container,
  Inner,
  Pane,
  Title,
  SubTitle,
  Image1,
  AnimationCard1,
  AnimationCard2,
} from './styles/jumbotron';

export default function Jumbotron({
  children,
  direction = 'row',
  ...restProps
}) {
  return (
    <Container {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Container>
  );
}

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumbotronSubTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image1 = ({ children, ...restProps }) => {
  return (
    <div style={{position: "relative", overflow: "hidden"}}>
      <Image1 {...restProps} />
      <AnimationCard1>
        <video autoPlay playsInline muted loop>
          <source
            src="https://assets.nflxext.com/ffe/siteui/acquisition/ourStory/fuji/desktop/video-tv-0819.m4v"
            type="video/mp4"
          ></source>
        </video>
      </AnimationCard1>
    </div>
  );
};
Jumbotron.Image2 = ({ children, ...restProps }) => {
  return (
    <div style={{ position: 'relative', overflow: 'hidden' }}>
      <Image1 {...restProps} />
      
    </div>
  );
};
Jumbotron.Image3 = ({ children, ...restProps }) => {
  return (  
    <div style={{ position: 'relative', overflow: 'hidden' }}>
      <Image1 {...restProps} />
      <AnimationCard2 style={{  overflow: 'hidden' }}>
        <video autoPlay playsInline muted loop>
          <source
            src="https://assets.nflxext.com/ffe/siteui/acquisition/ourStory/fuji/desktop/video-devices.m4v"
            type="video/mp4"
          ></source>
        </video>
      </AnimationCard2>
    </div>
  );
};