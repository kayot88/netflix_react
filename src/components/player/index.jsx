import React, { createContext, useState, useContext } from 'react';

import ReactDOM from 'react-dom';
import { Button, Close, Container, Inner, Overlay } from './styles/player';

export const PlayerContext = createContext();

export const Player = ({ children, ...restProps }) => {
  const [showPlayer, setShowPlayer] = useState(false);
  return (
    <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
      <Container {...restProps}>{children}</Container>
    </PlayerContext.Provider>
  );
};

Player.Video = function PlayerVideo({ src, ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);
  return showPlayer
    ? ReactDOM.createPortal(
        <Overlay {...restProps} data-testid="player">
          <Inner >
            <video id="netflix-player" controls>
              <source src={src} type="video/mp4" />
            </video>
            <Close role='close' onClick={() => setShowPlayer(false)} />
          </Inner>
        </Overlay>,
        document.body
      )
    : null;
};

Player.Button = function PlayerButton({ ...restProps }) {
  const { showPlayer, setShowPlayer } = useContext(PlayerContext);
  return (
    <Button onClick={() => setShowPlayer((showPlayer) => !showPlayer)}>
      Play
    </Button>
  );
};


