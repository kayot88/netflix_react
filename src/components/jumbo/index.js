import React from "react";
import { Container, Inner, Pane, Title, SubTitle, Image } from "./styles/jumbotron";

export const Jumbotron = ({ children, direction = "row", ...restProps }) => {
  return <Inner direction={direction}>{children}</Inner>;
};

Jumbotron.Container = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};
Jumbotron.Pane = ({ children, ...restProps }) => {
  return <Pane {...restProps}>{children}</Pane>;
};
Jumbotron.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};
Jumbotron.SubTitle = ({ children, ...restProps }) => {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
Jumbotron.Image = ({ ...restProps }) => {
  return <Image {...restProps} />;
};
