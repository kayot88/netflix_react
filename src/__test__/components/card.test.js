import { fireEvent, getByAltText, render } from '@testing-library/react';
import 'jest-styled-components';
import React from 'react';
import { Card, Player } from '../../components';

const category = 'series';

const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
        description: 'Tiger King description',
        id: '07adcf19-517f-41b4-8f8b-04fee694bce1',
        title: 'Tiger King',
        docId: 'IcRxJBanzE7nmIp2GMGb',
      },
    ],
  },
  {
    title: 'Feel Good',
    data: [
      {
        title: 'Juno',
        genre: 'feel-good',
        description: 'Juno description',
        maturity: '0',
        id: 'a938b5a6-fe25-4a06-8050-353bc7146ccd',
        slug: 'juno',
        docId: '4JDgdf5vE0K3YrW9ZnbP',
      },
    ],
  },
];

describe('<Card/> rendered with some data', () => {
  it('should render <card/>', () => {
    const {
      getByText,
      queryByText,
      debug,
      container,
      getByTestId,
      getByAltText,
    } = render(
      <Card>
        <Card.Group>
          {slideRows.map((slideItem) => (
            <Card key={`${category} - ${slideItem.title.toLowerCase()}`}>
              <Card.Title>{slideItem.title}</Card.Title>
              <Card.Entities>
                {slideItem.data.map((item) => {
                  return (
                    <Card.Item
                      key={item.docId}
                      item={item}
                      data-testid={`${item.slug}`}
                    >
                      <Card.Image
                        src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                      />
                      <Card.Meta>
                        <Card.SubTitle>{item.title}</Card.SubTitle>
                        <Card.Text>{item.description}</Card.Text>
                      </Card.Meta>
                    </Card.Item>
                  );
                })}
              </Card.Entities>
              <Card.Feature category={category}>
                <Player>
                  <Player.Button />
                  <Player.Video src="/videos/bunny.mp4" />
                </Player>
              </Card.Feature>
            </Card>
          ))}
        </Card.Group>
      </Card>
    );
    expect(getByText('Documentaries')).toBeTruthy();
    expect(getByText('Feel Good')).toBeTruthy();

    expect(queryByText('documentaries')).toBeFalsy();
    expect(queryByText('18')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render feature text', () => {
    const { queryByText, container, getByTestId, getByAltText } = render(
      <Card>
        <Card.Group>
          {slideRows.map((slideItem) => (
            <Card key={`${category} - ${slideItem.title.toLowerCase()}`}>
              <Card.Title>{slideItem.title}</Card.Title>
              <Card.Entities>
                {slideItem.data.map((item) => {
                  return (
                    <Card.Item
                      key={item.docId}
                      item={item}
                      data-testid={`${item.slug}`}
                    >
                      <Card.Image
                        src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                      />
                      <Card.Meta>
                        <Card.SubTitle>{item.title}</Card.SubTitle>
                        <Card.Text>{item.description}</Card.Text>
                      </Card.Meta>
                    </Card.Item>
                  );
                })}
              </Card.Entities>
              <Card.Feature category={category}>
                <Player>
                  <Player.Button />
                  <Player.Video src="/videos/bunny.mp4" />
                </Player>
              </Card.Feature>
            </Card>
          ))}
        </Card.Group>
      </Card>
    );
    fireEvent.click(getByTestId('tiger-king'));
    expect(queryByText('18')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('18')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
