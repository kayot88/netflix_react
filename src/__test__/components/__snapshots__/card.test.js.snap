// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> rendered with some data should render <card/> 1`] = `
.c4 {
  font-size: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #e5e5e5;
  font-weight: bold;
  margin-left: 56px;
  margin-right: 56px;
  margin-top: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 50px;
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  font-size: 12px;
  color: #fff;
  font-weight: bold;
  margin-top: 0;
  margin-bottom: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: none;
}

.c13 {
  margin-top: 5px;
  font-size: 10px;
  color: #fff;
  margin-bottom: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: none;
  line-height: normal;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c9 {
  display: none;
  position: absolute;
  bottom: 0;
  padding: 10px;
  background-color: #0000008f;
}

.c7 {
  border: 0;
  width: 100%;
  max-width: 305px;
  cursor: pointer;
  height: auto;
  padding: 0;
  margin: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: 5px;
  position: relative;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
}

.c6:hover {
  -webkit-transform: scale(1.3);
  -ms-transform: scale(1.3);
  transform: scale(1.3);
  z-index: 99;
}

.c6:first-of-type {
  margin-left: 56px;
}

.c6:last-of-type {
  margin-right: 56px;
}

@media (max-width:1000px) {
  .c1 > .c3 {
    margin-left: 30px;
  }
}

@media (min-width:1100px) {
  .c2 > .c0:first-of-type {
    margin-top: -150px;
  }
}

@media (min-width:1200px) {
  .c6:hover .c8,
  .c6 .c12,
  .c6 .c10 {
    display: block;
    z-index: 100;
  }
}

@media (max-width:1000px) {
  .c6:first-of-type {
    margin-left: 30px;
  }
}

@media (max-width:1000px) {
  .c6:last-of-type {
    margin-right: 30px;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c0 c1"
    >
      <p
        class="c3 c4"
      >
        Documentaries
      </p>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="tiger-king"
        >
          <img
            class="c7"
            src="/images/series/documentaries/tiger-king/small.jpg"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
            >
              Tiger King
            </p>
            <p
              class="c12 c13"
            >
              Tiger King description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0 c1"
    >
      <p
        class="c3 c4"
      >
        Feel Good
      </p>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="juno"
        >
          <img
            class="c7"
            src="/images/series/feel-good/juno/small.jpg"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
            >
              Juno
            </p>
            <p
              class="c12 c13"
            >
              Juno description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card/> rendered with some data should render feature text 1`] = `
.c4 {
  font-size: 24px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #e5e5e5;
  font-weight: bold;
  margin-left: 56px;
  margin-right: 56px;
  margin-top: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 50px;
  box-sizing: border-box;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  font-size: 12px;
  color: #fff;
  font-weight: bold;
  margin-top: 0;
  margin-bottom: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: none;
}

.c13 {
  margin-top: 5px;
  font-size: 10px;
  color: #fff;
  margin-bottom: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: none;
  line-height: normal;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c9 {
  display: none;
  position: absolute;
  bottom: 0;
  padding: 10px;
  background-color: #0000008f;
}

.c7 {
  border: 0;
  width: 100%;
  max-width: 305px;
  cursor: pointer;
  height: auto;
  padding: 0;
  margin: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: 5px;
  position: relative;
  cursor: pointer;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
}

.c6:hover {
  -webkit-transform: scale(1.3);
  -ms-transform: scale(1.3);
  transform: scale(1.3);
  z-index: 99;
}

.c6:first-of-type {
  margin-left: 56px;
}

.c6:last-of-type {
  margin-right: 56px;
}

@media (max-width:1000px) {
  .c1 > .c3 {
    margin-left: 30px;
  }
}

@media (min-width:1100px) {
  .c2 > .c0:first-of-type {
    margin-top: -150px;
  }
}

@media (min-width:1100px) {
  .bVgKno > .c0:first-of-type {
    margin-top: -150px;
  }
}

@media (min-width:1200px) {
  .c6:hover .c8,
  .c6 .c12,
  .c6 .c10 {
    display: block;
    z-index: 100;
  }
}

@media (max-width:1000px) {
  .c6:first-of-type {
    margin-left: 30px;
  }
}

@media (max-width:1000px) {
  .c6:last-of-type {
    margin-right: 30px;
  }
}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:1000px) {
  .QXdcK .c3 {
    font-size: 20px;
    line-height: 20px;
    margin-bottom: 10px;
  }
}

@media (max-width:1000px) {

}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c0 c1"
    >
      <p
        class="c3 c4"
      >
        Documentaries
      </p>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="tiger-king"
        >
          <img
            class="c7"
            src="/images/series/documentaries/tiger-king/small.jpg"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
            >
              Tiger King
            </p>
            <p
              class="c12 c13"
            >
              Tiger King description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c0 c1"
    >
      <p
        class="c3 c4"
      >
        Feel Good
      </p>
      <div
        class="c5"
      >
        <div
          class="c6"
          data-testid="juno"
        >
          <img
            class="c7"
            src="/images/series/feel-good/juno/small.jpg"
          />
          <div
            class="c8 c9"
          >
            <p
              class="c10 c11"
            >
              Juno
            </p>
            <p
              class="c12 c13"
            >
              Juno description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
